elixir/apps/web/test/web/live/resources/edit_test.exs

```diff
@@ -354,8 +328,41 @@ defmodule Web.Live.Resources.EditTest do
     assert saved_resource = Repo.get_by(Domain.Resources.Resource, id: resource.id)
     assert saved_resource.name == attrs.name
 
-    assert saved_resource.filters == [
-             %Domain.Resources.Resource.Filter{protocol: :all, ports: []}
-           ]
+    assert saved_resource.filters == []
+  end
+
+  test "prevents saving resource if traffic filters set when traffic filters disabled", %{
+    account: account,
+    group: group,
+    identity: identity,
+    resource: resource,
+    conn: conn
+  } do
+    Domain.Config.feature_flag_override(:traffic_filters, false)
+
+    attrs = %{
+      name: "foobar.com",
+      filters: %{
+        icmp: %{enabled: true},
+        tcp: %{ports: "8080, 4443"},
+        udp: %{ports: "4000 - 5000"}
+      }
+    }
+
+    {:ok, lv, _html} =
+      conn
+      |> authorize_conn(identity)
+      |> live(~p"/#{account}/resources/#{resource}/edit?site_id=#{group}")
+
+    # ** (ArgumentError) could not find non-disabled input, select or textarea with name "resource[filters][tcp][ports]" within:
```

Comment:
This test doesn't actually test that traffic filters can't be saved but that fields are disabled if it's set to false. And if that is what we want to test then let's rename the test and just use `assert has_element?(lv, "...[disabled]")`

Reply:
👍 Updated